jupyterhub/templates/_helpers-netpol.tpl

```diff
@@ -0,0 +1,73 @@
+{{- define "jupyterhub.networkPolicy.standardEgressConfiguration" -}}
+{{- include "jupyterhub.networkPolicy.standardEgressConfiguration.withLeadingNewLine" . | trimPrefix "\n" }}
+{{- end }}
+
+{{- define "jupyterhub.networkPolicy.standardEgressConfiguration.withLeadingNewLine" -}}
+{{- $root := index . 0 }}
+{{- $netpol := index . 1 }}
+{{- if and ($netpol.egressAllowRules.dnsPortsPrivateIPs) (not $netpol.egressAllowRules.privateIPs) }}
+# Allow outbound connections to the DNS port in the private IP ranges
+- ports:
+    - protocol: UDP
+      port: 53
+    - protocol: TCP
+      port: 53
+  to:
+    - ipBlock:
+        cidr: 10.0.0.0/8
+    - ipBlock:
+        cidr: 172.16.0.0/12
+    - ipBlock:
+        cidr: 192.168.0.0/16
+{{- end }}
+
+{{- if $netpol.egressAllowRules.nonPrivateIPs }}
+# Allow outbound connections to non-private IP ranges
+{{- if $netpol.egressAllowRules.privateIPs }}
+# Allow outbound connections to private IP ranges (and their DNS ports)
+{{- end }}
+{{- if $netpol.egressAllowRules.cloudMetadataServer }}
+# Allow outbound connections to cloud metadata server
+{{- end }}
+- to:
+    - ipBlock:
+        cidr: 0.0.0.0/0
+        {{- if or (not $netpol.egressAllowRules.privateIPs) (not $netpol.egressAllowRules.cloudMetadataServer) }}
+        except:
+          {{- if not $netpol.egressAllowRules.privateIPs }}
+          # Don't allow outbound connections to private IP ranges
```

Comment:
Link to the RFC that lists these as the private ranges? With bare CIDRs like this I find it very helpful to figure out where they came from.

Reply:
I have the link in the schema.yaml / config reference. If i mention the definition of what a private IP is wherever it is mentioned it would be mentioned in too many places and I would see it as cluttering the other logic, so I think mentioning it here isn't justified.

In practice, I've opted to have each rule emit ~one line of comments about it being active (or not active in this case), and referred to the schema.yaml / config reference for a centralized and more in depth explanation.

Reply:
I generally agree, but I only want it exactly where the IPs themselves are listed - otherwise they look like [magic numbers](https://refactoring.guru/replace-magic-number-with-symbolic-constant). So this isn't a definition of what private IPs are, but a reference to exactly where these hard coded CIDR values come from.